### PR TITLE
Refresh plugin build for March 2024 and Java 17/21

### PR DESCRIPTION
**Pull request by basil**
- Run build on Java 17/21, conforming to latest archetype
- Update to latest plugin parent POM and BOM
- Slim down JAR by using library plugins where possible

**Tesing done:** Successful CI build at https://ci.jenkins.io/job/Plugins/job/pipeline-aws-plugin/job/PR-322/1/

### CODE REVIEW

File: .mvn/extensions.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<extensions xmlns="http://maven.apache.org/EXTENSIONS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/EXTENSIONS/1.0.0 http://maven.apache.org/xsd/core-extensions-1.0.0.xsd">
+  <extension>
+    <groupId>io.jenkins.tools.incrementals</groupId>
+    <artifactId>git-changelist-maven-extension</artifactId>
+    <version>1.7</version>
+  </extension>
+</extensions>
```
Comment (basil) [inline .mvn/extensions.xml:7]: See https://github.com/jenkinsci/incrementals-tools.

File: .mvn/maven.config [hunk 2]
```diff
@@ -0,0 +1,2 @@
+-Pconsume-incrementals
+-Pmight-produce-incrementals
```
Comment (basil) [inline .mvn/maven.config:2]: See https://github.com/jenkinsci/incrementals-tools.

File: Jenkinsfile [hunk 3]
```diff
@@ -5,6 +5,6 @@
 buildPlugin(
   useContainerAgent: true,
   configurations: [
-    [platform: 'linux', jdk: 11], // should be bumped to 21 before Nov 2024
+    [platform: 'linux', jdk: 21],
```
Comment (basil) [inline Jenkinsfile:8]: See https://github.com/jenkinsci/archetypes.

File: pom.xml [hunk 4]
```diff
@@ -4,16 +4,16 @@
     <parent>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>plugin</artifactId>
-        <version>4.72</version>
+        <version>4.80</version>
```
Comment (basil) [inline pom.xml:7]: Latest version at the time of this writing.

File: pom.xml [hunk 5]
```diff
@@ -4,16 +4,16 @@
...
         <relativePath />
     </parent>
     <groupId>de.taimos</groupId>
     <artifactId>pipeline-aws</artifactId>
-    <version>1.45-SNAPSHOT</version>
+    <version>${revision}${changelist}</version>
```
Comment (basil) [inline pom.xml:12]: See https://github.com/jenkinsci/incrementals-tools.

File: pom.xml [hunk 6]
```diff
@@ -4,16 +4,16 @@
...
     <packaging>hpi</packaging>
     <name>Pipeline: AWS Steps</name>
     <scm>
-        <connection>scm:git:https://github.com/${gitHubRepo}</connection>
+        <connection>scm:git:https://github.com/${gitHubRepo}.git</connection>
```
Comment (basil) [inline pom.xml:16]: See https://github.com/jenkinsci/archetypes.

File: pom.xml [hunk 7]
```diff
@@ -22,8 +22,8 @@
         <dependencies>
             <dependency>
                 <groupId>io.jenkins.tools.bom</groupId>
-                <artifactId>bom-2.401.x</artifactId>
-                <version>2675.v1515e14da_7a_6</version>
+                <artifactId>bom-2.414.x</artifactId>
+                <version>2928.ved44ea_84e034</version>
```
Comment (basil) [inline pom.xml:26]: See https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/

File: pom.xml [hunk 8]
```diff
@@ -35,7 +35,7 @@
             <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
             <distribution>repo</distribution></license>
     </licenses>
-    <url>https://github.com/jenkinsci/pipeline-aws-plugin</url>
+    <url>https://github.com/jenkinsci/${project.artifactId}-plugin</url>
```
Comment (basil) [inline pom.xml:38]: See https://github.com/jenkinsci/archetypes.

File: pom.xml [hunk 9]
```diff
@@ -76,30 +76,13 @@
...
-                  <enforceBytecodeVersion>
-                    <excludes combine.children="append">
-                      <exclude>org.projectlombok:lombok</exclude>
-                    </excludes>
-                  </enforceBytecodeVersion>
-                </rules>
-              </configuration>
-            </execution>
-          </executions>
-      </plugin>
```
Comment (basil) [inline pom.xml:102]: No longer needed with latest Lombok.

**Comment by MarkEWaite** [comment]
Changes all look correct to me

**Review by gounthar — APPROVED** [review]
Thanks a lot for the hard work, Basil! 🚀

**Review by hoegertn — APPROVED** [review]

**Comment by basil** [comment]
Thanks @hoegertn! Can this please be released?

**Comment by hoegertn** [comment]
Will do later today

**Comment by hoegertn** [comment]
Released as 1.45

Hope everything works fine.

**Comment by gounthar** [comment]
Thanks a lot for your reactivity. 👍

**Comment by basil** [comment]
Thank you very much!

**Pull request merged**